handlers/public.go

```diff
@@ -0,0 +1,208 @@
+	Resource    string `json:"resource"`
+	State       string `json:"state"`
+}
+
+// PaymentResource is the payment details to be stored in the DB and returned in the response
+type PaymentResource struct {
+	//	ID                      string    `json:"id"                                  bson:"_id"`                                 // TODO implement
+	Amount                  string    `json:"amount"                              bson:"amount"`                              // TODO implement
+	AvailablePaymentMethods []string  `json:"available_payment_methods,omitempty" bson:"available_payment_methods,omitempty"` // TODO implement
+	CompletedAt             time.Time `json:"completed_at,omitempty"              bson:"completed_at,omitempty"`              // TODO implement
+	CreatedAt               time.Time `json:"created_at,omitempty"                bson:"created_at,omitempty"`
+	CreatedBy               CreatedBy `json:"created_by"                          bson:"created_by"`
+	Description             string    `json:"description"                         bson:"description"`              // TODO implement
+	Links                   Links     `json:"links"                               bson:"links"`                    // TODO implement
+	PaymentMethod           string    `json:"payment_method,omitempty"            bson:"payment_method,omitempty"` // TODO implement
+	Reference               string    `json:"reference,omitempty"                 bson:"reference,omitempty"`      // TODO implement
+	Status                  string    `json:"status"                              bson:"status"`                   // TODO implement
+}
+
+// CreatedBy is the user who is creating the payment session
+type CreatedBy struct {
+	Email    string `bson:"email"`
+	Forename string `bson:"forename"`
+	ID       string `bson:"id"`
+	Surname  string `bson:"surname"`
+}
+
+// Links is a set of URLs related to the resource, including self
+type Links struct {
+	Journey  string `json:"journey"`
+	Resource string `json:"resource"`
+	Self     string `json:"self"`
+}
+
+//Data is a representation of the top level data retrieved from the Transaction API
+type Data struct {
+	CompanyName string            `json:"company_name"`
+	Filings     map[string]Filing `json:"filings"`
+}
+
+//Filing is a representation of the Filing subsection of data retrieved from the Transaction API
+type Filing struct {
+	Description string `json:"description"`
+}
+
+// Register defines the route mappings
+func Register(r *pat.Router) {
+	r.Get("/healthcheck", getHealthCheck).Name("get-healthcheck")
+	r.Post("/payments", createPaymentSession).Name("create-payment")
+}
+
+func getHealthCheck(w http.ResponseWriter, r *http.Request) {
+	w.WriteHeader(http.StatusOK)
+}
+
+var client http.Client
+
+func createPaymentSession(w http.ResponseWriter, req *http.Request) {
+	if req.Body == nil {
+		log.ErrorR(req, fmt.Errorf("request body empty"))
+		w.WriteHeader(http.StatusBadRequest) // 400
```

Comment:
The comment isn't needed here. The GoDoc should contain the status code represented by http.StatusBadRequest.